### PR TITLE
philip_pal: add main for CLI mode

### PR DESCRIPTION
**Pull request by smlng**
This PR primarily adds CLI mode to the philip_pal package, that means you can now run it as a single command within a CLI/shell. To test this and see available commands run:

```
python3 -m philip_pal --help
```

which should give you 

```
usage: python3 -m philip_pal [-h] [--verbose] [--dut_reset] [--mcu_reset]
                             [port]

positional arguments:
  port         PHiLIP serial port

optional arguments:
  -h, --help   show this help message and exit
  --verbose    Enable more output
  --dut_reset  Reset device-under-test (DUT)
  --mcu_reset  Reset PHiLIP MCU
```

### CODE REVIEW

**Comment by smlng** [comment]
and piggy backed a renaming of `reset_mcu` to `mcu_reset`

File: IF/philip_pal/philip_pal/__main__.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#
+# This file is subject to the terms and conditions of the MIT License. See the
+# file LICENSE in the top level directory for more details.
+# SPDX-License-Identifier:    MIT
+"""packet init for PHiLIP PAL
+This exposes a main function to run the packet from CLI
+"""
+from .philip_if import PhilipExtIf as Phil
+
+if __name__ == '__main__':
```
Comment (smlng) [inline IF/philip_pal/philip_pal/__main__.py]: note to self: this should be redundant

**Comment by smlng** [comment]
@MrKevinWeiss: I removed the renaming of `reset_mcu` this should go into a separate PR. Further I now use the logging module for verbose output as you suggested. I also fixed any warning by pylint and flake8 so this up to standards now 😄

**Comment by MrKevinWeiss** [comment]
...Do we have to bump the version or should we collect things until ready for a new pip dist?

**Comment by MrKevinWeiss** [comment]
pip package is updated, should be available with `pip3 install philip_pal`

**Pull request merged**